integration-tests/utils/integration-utils.js

```diff
@@ -66,17 +66,38 @@ const getTableCellPath = (tableName, row, column) => {
   return `//*[@role="table" and @aria-label="${tableName}"]//*[@role="row"][${row}]//*[@role="cell"][${column}]`
 }
 
-const getTableHeaderPath = (tableName, column) => {
-  return `//*[@role="table" and @aria-label="${tableName}"]//*[@role="row"][1]//*[@role="columnheader"][${column}]`
+const getTableColIndex = async (page, { tableName, columnHeader }) => {
+  const colHeaderNode = await findElement(page, `//*[@role="table" and @aria-label="${tableName}"]//*[@role="columnheader" and @aria-colindex and contains(normalize-space(.),"${columnHeader}")]`)
+  return page.evaluate(node => node.getAttribute('aria-colindex'), colHeaderNode)
 }
 
-const findTableCellText = async (page, path, textContains, options) => {
-  const xpath = `${path}[contains(normalize-space(.),"${textContains}")]`
-  return (await page.waitForXPath(xpath, options))
+const getTableTextWithinColumn = async (page, { tableName, columnHeader, textContains, isDescendant = false }) => {
+  const colIndex = await getTableColIndex(page, { tableName, columnHeader })
+  const baseXpath = `//*[@role="table" and @aria-label="${tableName}"]//*[@role="row"]//*[@role="cell" and @aria-colindex = "${colIndex}"]`
+  const xpath = `${baseXpath}${isDescendant ? '//*' : ''}[contains(normalize-space(.),"${textContains}")]`
+  return xpath
 }
 
-const clickTableCell = async (page, tableName, row, column, options) => {
-  const tableCellPath = getTableCellPath(tableName, row, column)
+const assertRowMatches = async (page, { tableName, expectedColumnValues, withKey: { column, textContains } }) => {
```

Comment:
nit: Could you rename this to `assertRowHas`?

```suggestion
const assertRowHas = async (page, { tableName, expectedColumnValues, withKey: { column, textContains } }) => {
```